csrc/mmdeploy/net/rknn/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+project(mmdeploy_rknn_net)
+
+add_library(rknn SHARED IMPORTED)
+
+if(DEFINED ENV{RKNPU2_DIR})
+    file(TO_CMAKE_PATH $ENV{RKNPU2_DIR} RKNPU2_DIR)
+else()
+    message(FATAL_ERROR "RKNPU2_DIR env must be defined")
+endif()
+
+set_target_properties(rknn PROPERTIES
+  IMPORTED_LOCATION "${RKNPU2_DIR}/runtime/RK3588/Linux/librknn_api/aarch64/librknnrt.so"
```

Comment:
I found in `rknpu2` runtime directory, the structure is as following:
```
.
├── Android.bp
├── Android.go
├── init.rknn_server.rc
├── RK356X
├── RK3588
└── RV1106
```
It means that we have to link the corresponding library according to the target device model. But here we hard code the RK3588's runtime path. 
One way is to set the rockchip model in the toolchain file, like `set(CMAKE_SYSTEM_PROCESSOR RK3588)`. 
And we can `set_target_properties` according to `${CMAKE_SYSTEM_PROCESSOR}`

Reply:
How about `add_subdirectory(rknn)` unconditionally, and
```cmake
set(SUPPORTED_MODELS RK3588 RK356X RV1126)

foreach (_model_name IN LISTS SUPPORTED_MODELS)
    if (_model_name IN_LIST MMDEPLOY_TARGET_BACKENDS)
        set(MODEL_NAME ${_model_name})
        break ()
    endif ()
endforeach ()

if (MODEL_NAME)
    # add targets
endif ()
```

Reply:
you mean -DMMDEPLOY_TARGET_BACKENDS="RK3588", right?
Looks better than `set(CMAKE_SYSTEM_PROCESSOR RK3588)`.
I checked https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html, 
`CMAKE_SYSTEM_PROCESSOR`refers to the cpu architecture.

Reply:
@lzhangzz if we use RK3588, PR356X as MMDEPLOY_TARGET_BACKENDS arguments, 
exporting SDK model meta info will suffer a lot

Reply:
Yes, some codes need to be changed. BTW, why can't we add new flags like `MMDEPLOY_TARGET_DEVICE`?

Reply:
After reconsidering it, `Device` in mmdeploy SDK means it has `buffer`, `stream`, and `event` resources. But npu doesn't have any such concepts. So seems not appropriate to put RK3588 in MMDEPLOY_TARGET_DEVICES. 
In rockchip case, the precise description is device(CPU), backend(rknn), rockchip model(rk3588, ...)
But if we add another build option ROCKCHIP_MODEL, that might be a burden for users.
So I prefer `-DMMDEPLOY_TARGET_BACKENDS="RK3588"`.